### PR TITLE
ENH: added option to read rainbow files from file handle

### PR DESCRIPTION
**Pull request by cchwala**
As discussed, [here](https://github.com/cchwala/wradlib/commit/d55678fda5afc9945844f258586d6a47da68e4f3#commitcomment-21557664), this is now the PR to make it possible to read Rainbow files from a file handle, e.g. when reading from a zipped archive.

I assumed that the low level functions do not need the `apichange_kwarg` since nobody ever uses them directly and they only get called by `read_Rainbow`. It would be hard (or impossible without writing to a temp file) to transform the `fid` file handles into the now deprecated `filename` strings because in many cases (or at least in my application) there is no path to the files which e.g. may belong to a zipped archive.

After rebasing on master I cannot run the tests. They fail with "The process has forked and you cannot use this CoreFoundation functionality safely. You MUST exec().". Hence, I hope that the CI runs will complete.

### CODE REVIEW

**Comment by cchwala** [comment]
Sorry for the iterative fixing based on CI errors. As stated above, test do currently not run locally on my machine. Feel free to squash the commits. The first one is labled wrongly anyway. "WIP" should now maybe be "ENH", when it's finally working...

**Comment by kmuehlbauer** [comment]
> After rebasing on master I cannot run the tests. They fail with "The process has forked and you cannot use this CoreFoundation functionality safely. You MUST exec().". Hence, I hope that the CI runs will complete.

@cchwala Are you on OSX? On my linux box everything runs fine, when I start the tests on the console with ```python testrunner.py -a```. 
The changes were introduced with #139. We were forced to run testsuites in subprocesses to overcome the overall memory limit of 3GB on travis-ci.

**Comment by cchwala** [comment]
@kmuehlbauer Yes OSX. I will try on a Linux machine. Should I open an issue concerning the tests on OSX?

**Comment by kmuehlbauer** [comment]
@cchwala This is a problem obviously with OSX (and maybe with windows). I found this: http://stackoverflow.com/questions/16254191/python-rpy2-and-matplotlib-conflict-when-using-multiprocessing

But you can try `xvfb` to make the tests headless. See `.travis.yml`. 

You can open an issue, and we can reference everything there.

**Comment by cchwala** [comment]
@kmuehlbauer Sorry, I am very busy this week. Would it suffice to finish this PR next monday?

**Comment by kmuehlbauer** [comment]
@cchwala I would really like to roll out wradlib release `0.10.0.` on next monday. I could pull your additions from your fork and add my suggested changes. Are you OK with this way of working?

**Comment by cchwala** [comment]
@kmuehlbauer Okay, I will do it till Friday evening/night. It does not look like too much work anyway and since I started this I also want to bring it to an end ;-).

**Comment by kmuehlbauer** [comment]
@cchwala This would fit. 

@heistermann Regarding the **apichange**. Since we do not have something for `*args` we do not have much options. From the workflow it seems unnecessary to open/close the file multiple times. So I tend to silently break the API for the low-level functions (leave the `get_RB_blob_from_file` handle both `filename` and `fid`). Adding `.. versionchanged:: 0.10.0` in all changed functions should suffice.

**Comment by cchwala** [comment]
@kmuehlbauer #137 produces a diff for `io.py` that is the whole file content. I have a hard time to solve the merge conflict since its 2000 lines of code vs 2000 lines of code. Right now I am assuming that you only changed things concerning `io.write_raster_dataset` within `io.py` and will manually resolve the conflicts. Any additional info from your side?

**Comment by cchwala** [comment]
@kmuehlbauer Merge done. Test should complete, at least they now do on my Mac. Now on to the requested changes.

File: wradlib/io.py [hunk 1]
```diff
@@ -1616,53 +1610,51 @@ def get_RB_blobs_from_file(filename, rbdict):
...
+        raise IOError('Could not read from file handle')
 
     xmltodict = util.import_optional('xmltodict')
 
     return xmltodict.parse(header)
 
 
-def read_Rainbow(filename, loaddata=True):
-    """"Reads Rainbow files files according to their structure
+@util.apichange_kwarg('0.9.2', par='filename', typ=str, expar='f')
```
Comment (cchwala) [inline wradlib/io.py]: Removed the decorator according to your request. But isn't the decorator also mean to catch the case where people have `read_Rainbow(filename='my_file.azi')` in their code?

Comment (kmuehlbauer) [inline wradlib/io.py]: *filename* is no *kwarg*, so it isn't catched by this decorator. We would need some decorator for *args*.

**Comment by kmuehlbauer** [comment]
@cchwala Sorry for the inconvenience with the io.py. I remember having issues with the line endings. That's probably the cause for the large difference.

**Comment by kmuehlbauer** [comment]
@cchwala Thanks for your contribution! Hoping to see more from your side 👍.

**Pull request merged**